napari/layers/base/base.py

```diff
@@ -470,28 +500,30 @@ def thumbnail(self):
         return self._thumbnail
 
     @thumbnail.setter
-    def thumbnail(self, thumbnail):
-        if 0 in thumbnail.shape:
-            thumbnail = np.zeros(self._thumbnail_shape, dtype=np.uint8)
-        if thumbnail.dtype != np.uint8:
+    def thumbnail(self, value):
```

Comment:
To be honest I would revert these. I think it should be `image` or `thumbnail`. If you *really* care about making it value for consistency with events, then the first thing I would do is `image = value` and proceed from that. `value is just so generic, and the rest of the code makes no sense without context.